### PR TITLE
Allow null parent to override unrecognized parents

### PR DESCRIPTION
**Pull request by PhantomWatson**
Moving pull request https://github.com/cakephp/cakephp/pull/6151 into 2.7 branch.

### CODE REVIEW

**Comment by markstory** [comment]
Looks like other changes from 2.6 came in as well. That's ok, as we have to do that merge anyways.

**Comment by robertpustulka** [comment]
Next time try cherry-picking your commits to another branch :).

**Comment by PhantomWatson** [comment]
Excellent advice.

**Comment by dereuromark** [comment]
Shall we continue with this? Does it have a test case?

**Comment by PhantomWatson** [comment]
Yep, [ModelReadTest.php has been modified](https://github.com/cakephp/cakephp/pull/6151/files) to reflect the new behavior.

File: lib/Cake/Network/CakeResponse.php [hunk 1]
```diff
@@ -68,6 +68,7 @@ class CakeResponse {
 		415 => 'Unsupported Media Type',
 		416 => 'Requested range not satisfiable',
 		417 => 'Expectation Failed',
+		429 => 'Too Many Requests',
```
Comment (lorenzo) [inline lib/Cake/Network/CakeResponse.php]: Why is this change here?

Comment (ADmad) [inline lib/Cake/Network/CakeResponse.php]: @lorenzo He had made the commits on top of 2.6 branch and the PR is opened against 2.7, that might be the reason of the wacky diff.

Comment (PhantomWatson) [inline lib/Cake/Network/CakeResponse.php]: Yeah, I had a newb moment and didn't cherry-pick my commits. I can re-do the pull request if it would make things easier.

Comment (lorenzo) [inline lib/Cake/Network/CakeResponse.php]: I would prefer if you do that (you can rebase and then force push)

Comment (PhantomWatson) [inline lib/Cake/Network/CakeResponse.php]: Done.

**Comment by markstory** [comment]
Closing as this branch can no longer be merged. If you have time to update it, please reopen the pull request :smile:.
